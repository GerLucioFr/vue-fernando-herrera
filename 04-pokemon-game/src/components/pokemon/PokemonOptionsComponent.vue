<template>
  <div class="options-container">
    <button
      v-for="option in pokemonOptions"
      :key="option.id"
      :disabled="chosenPokemon"
      @click="$emit('selection', option)"
    >
      {{ option.name }}
    </button>
  </div>
</template>

<script>
export default {
  name: 'PokemonOptionsComponent',
  props: {
    pokemonOptions: {
      type: Array,
      required: true,
    },
    chosenPokemon: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style scoped>
button {
  background-color: rgb(0, 95, 150);
  color: white;
}

button:hover {
  background-color: rgb(0, 50, 78);
}

button:disabled {
  background-color: rgb(214, 214, 214);
  color: black;
}

.options-container {
  display: flex;
  justify-content: center;
  gap: 16px;

  margin-top: 32px;
}

@media only screen and (max-width: 512px) {
  .options-container {
    flex-direction: column;
  }

  button {
    margin: 0 8px;
  }
}
</style>
