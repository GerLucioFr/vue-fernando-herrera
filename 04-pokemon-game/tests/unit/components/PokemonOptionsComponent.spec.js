import { shallowMount } from '@vue/test-utils';
import PokemonOptionsComponent from '@/components/pokemon/PokemonOptionsComponent';
import { pokemonOptionsMock } from '../mocks/pokemons.mock';

describe('Pokemon Options Component Test', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptionsComponent, {
      props: {
        pokemonOptions: pokemonOptionsMock,
        chosenPokemon: false,
      },
    });
  });

  it('should to match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should to show pokemons options in buttons', () => {
    const buttons = wrapper.findAll('button');

    expect(buttons).toHaveLength(4);
    buttons.forEach((button, index) => {
      expect(button.text()).toBe(pokemonOptionsMock[index].name);
    });
  });

  it('should to emit selected pokemon on click', async () => {
    const buttons = wrapper.findAll('button');

    buttons.forEach((button, index) => {
      button.trigger('click');
      const emited = wrapper.emitted('selection')[index][0];
      expect(emited).toStrictEqual(pokemonOptionsMock[index]);
    });
  });
});
