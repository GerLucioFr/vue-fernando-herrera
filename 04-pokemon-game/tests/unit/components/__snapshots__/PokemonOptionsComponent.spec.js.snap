// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Options Component Test should to match with snapshot 1`] = `<div class="options-container"><button>buneary</button><button>lanturn</button><button>minccino</button><button>dugtrio</button></div>`;
